artifactory.py

```diff
@@ -1056,6 +1028,10 @@ def deploy(
             headers["X-Explode-Archive"] = "true"
         if explode_archive_atomic:
             headers["X-Explode-Archive-Atomic"] = "true"
+        if by_checksum:
+            headers["X-Checksum-Deploy"] = "true"
+            if checksum:
```

Comment:
then this one could be removed, see comment for `deploy_by_checksum`

also, if this header is set, just check `if body: raise`

Reply:
Regarding `checksum `, see comment for `deploy_by_checksum`

Error should be raised if both `by_checksum` and `fobj` are not None. Good suggestion.